spec/factories/miq_server.rb

```diff
@@ -1,18 +1,19 @@
 FactoryGirl.define do
   factory :miq_server do
+    guid            { MiqUUID.new_guid }
+    zone            { FactoryGirl.build(:zone) }
     sequence(:name) { |n| "miq_server_#{seq_padded_for_sorting(n)}" }
     last_heartbeat  { Time.now.utc }
     status          "started"
     started_on      { Time.now.utc }
     stopped_on      ""
     version         '9.9.9.9'
-  end
-
-  factory :miq_server_master, :parent => :miq_server do
-    is_master      true
-  end
 
-  factory :miq_server_not_master, :parent => :miq_server do
-    is_master      false
+    factory :my_miq_server do
+      after(:create) do |ms|
+        MiqServer.stub(:my_guid).and_return(ms.guid)
+        MiqServer.my_server_clear_cache
```

Comment:
how do you propose we stub my_guid? Maybe the `MiqServer.my_server_clear_cache` is never necessary? I can test